numpy/core/tests/test_dtype.py

```diff
@@ -824,6 +821,53 @@ class Union(ctypes.Union):
         ))
         self.check(Union, expected)
 
+    def test_union_with_struct_endian_packed(self):
+        class Struct(ctypes.Structure):
+            _fields_ = [
+                ('one', ctypes.c_uint8),
+                ('two', ctypes.c_uint32.__ctype_le__)
+            ]
+            _pack_ = 1
+        class Union(ctypes.Union):
+            _fields_ = [
+                ('a', ctypes.c_uint8),
+                ('b', ctypes.c_uint16),
+                ('c', ctypes.c_uint32),
+                ('d', Struct),
+            ]
+        expected = np.dtype(dict(
+            names=['a', 'b', 'c', 'd'],
+            formats=['u1', '<u2', '<u4', [('one', 'u1'), ('two', '<u4')]],
+            offsets=[0, 0, 0, 0],
+            itemsize=ctypes.sizeof(Union)
+        ))
+        self.check(Union, expected)
+
+
+    def test_union_packed(self):
+        class Struct(ctypes.Structure):
+            _fields_ = [
+                ('one', ctypes.c_uint8),
+                ('two', ctypes.c_uint32.__ctype_le__)
```

Comment:
I'm not sure that `__ctype_le__` is worth having here - it's a no-op on little endian machines anyway, and is orthogonal to this test anyway.

Reply:
Fixed.